addons/web/static/tests/helpers/mock_server.js

```diff
@@ -1578,6 +1649,16 @@ export class MockServer {
                     }
                     criterion = [criterion[0], "in", childIDs];
                 }
+                // In case of many2many field, if domain operator is '=' generally change it to 'in' operator
+                const field = model.fields[criterion[0]] || {};
+                if (field.type === "many2many" && criterion[1] === "=") {
+                    if (criterion[2] === false) {
+                        // if undefined value asked, domain.js require equality with empty array
+                        criterion = [criterion[0], "=", []];
```

Comment:
why not in?

Reply:
https://github.com/odoo/odoo/blob/94500b18365674e8ca8be68abee1918330efb553/addons/web/static/src/core/domain.js#L214

```js
[].some(a=>[].includes(a)) // returns false
```

and for the legacy mock server, it uses [_.intersection](http://underscorejs.org/#intersection)
https://github.com/odoo/odoo/blob/94500b18365674e8ca8be68abee1918330efb553/addons/web/static/src/legacy/js/core/domain.js#L106

Reply:
[(field_name, "=", false)] won't work for many2many, in domain.js one returns fieldValue === value;

Reply:
that is why the criterion[2] part is not used here ;-)